### DISCUSSION TITLE
Rewrite order

### DISCUSSION THREAD

**Discussion by madsbuch** [vercel/vercel#3912] [General]
Hi there!

I have a deployment on a monorepo whre I am trying to to employ two rewrite rules. However, it seems like the system tries to apply the most general rule first, instead of the first one listed (json arrays are semantically ordered).

The code is as follows:

```json
"rewrites": [
    { "source": "/s/_next(.*)", "destination": "/apps/some-app/_next$1" },
    { "source": "(.*)", "destination": "/apps/some-app$1" }
  ],
```

When the second rule is put in, the first one ceases to work.

Does anyone have some context on this?

**Comment by mcsdevv** [comment]
Hey @madsbuch, this doesn't sound quite right. We'll take a look and get back to you on this.
